packager/mpd/base/xml/xml_node.cc

```diff
@@ -427,6 +431,46 @@ bool RepresentationXmlNode::AddVODOnlyInfo(const MediaInfo& media_info) {
     RCHECK(AddChild(std::move(child)));
   }
 
+  if (use_segment_list) {
+    XmlNode child("SegmentList");
+    if (media_info.has_reference_time_scale()) {
+      RCHECK(child.SetIntegerAttribute("timescale",
+                                       media_info.reference_time_scale()));
+    }
+
+    if (media_info.has_reference_time_scale()) {
+      RCHECK(child.SetIntegerAttribute("timescale",
+                                       media_info.reference_time_scale()));
+
+      if (media_info.has_media_duration_seconds()) {
+        const uint64_t duration_seconds =
+            (uint64_t) (target_segment_duration * media_info.reference_time_scale());
+        RCHECK(child.SetIntegerAttribute("duration", duration_seconds));
+      }
+    }
+
+    if (media_info.has_init_range()) {
+      XmlNode initialization("Initialization");
+      RCHECK(initialization.SetStringAttribute(
+          "range", RangeToString(media_info.init_range())));
+
+      RCHECK(child.AddChild(std::move(initialization)));
+    }
+
+    for (int i = 0; i < media_info.subsegment_ranges_size(); ++i) {
+      XmlNode subsegment("SegmentURL");
+      shaka::Range& subsegment_range =
+          const_cast<Range&>(media_info.subsegment_ranges(i));
+
+      RCHECK(subsegment.SetStringAttribute(
+          "mediaRange", RangeToString(subsegment_range)));
+
+      RCHECK(child.AddChild(std::move(subsegment)));
+    }
+
+    RCHECK(AddChild(std::move(child)));
+  }
+
```

Comment:
See the comments above, suggest combining it with the code block above to avoid code duplication